es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java

```diff
@@ -237,15 +249,13 @@ static S3ClientSettings getClientSettings(final Settings settings, final String
     static S3ClientSettings getClientSettings(final RepositoryMetaData metadata, final AWSCredentials credentials) {
         final Settings.Builder builder = Settings.builder();
         for (final String key : metadata.settings().keySet()) {
-            builder.put(PREFIX + "provided" + "." + key, metadata.settings().get(key));
+            builder.put(DEFAULT_PREFIX + key, metadata.settings().get(key));
         }
-        return getClientSettings(builder.build(), "provided", credentials);
+        return getClientSettings(builder.build(), "", credentials);
     }
 
-    private static <T> T getConfigValue(Settings settings, String clientName,
-                                        Setting.AffixSetting<T> clientSetting) {
-        final Setting<T> concreteSetting = clientSetting.getConcreteSettingForNamespace(clientName);
-        return concreteSetting.get(settings);
+    private static <T> T getConfigValue(Settings settings, String clientName, Setting<T> clientSetting) {
```

Comment:
unused clientName can be removed from method signature

Reply:
btw, i was wondering, is there a reason why we always use `default` client and do not allow to configuration of more than 1 clients ? Shall we follow the same pattern for the azure repo settings (using just one `default` client)?

Reply:
@kovrus we've never exposed the possibility to use different client configurations. We could do that in future but that is out of scope of this PR.
+1 for following the same on azure and implement multiple client support dedicated if needed.

Reply:
yes, I also agree to follow the same direction for the azure repo

Reply:
okey, thanks 👍 will limit that than.